doc/link_event_damping/Link-event-damping-HLD.md

```diff
@@ -0,0 +1,471 @@
+{
+    /**
+     * In this mode, operational status of hostif must be updated by application
+     * using hostif API with SAI_HOSTIF_ATTR_OPER_STATUS attribute. SAI adapter
+     * should not update the hostif operational status internally.
+     */
+
+
+    SAI_SWITCH_HOSTIF_OPER_STATUS_UPDATE_MODE_APPLICATION = 0,
+    /**
+     * In this mode, operational status of hostif is updated internally by SAI
+     * adapter. Update of hostif operational status by application using hostif
+     * API with SAI_HOSTIF_ATTR_OPER_STATUS is ignored.
+     */
+    SAI_SWITCH_HOSTIF_OPER_STATUS_UPDATE_MODE_SAI_ADAPTER = 1,
+} sai_switch_hostif_oper_status_update_mode_t;
+```
+
+For this proposal, we require vendor SAI to be running in `SAI_SWITCH_HOSTIF_OPER_STATUS_UPDATE_MODE_APPLICATION` mode to ensure update of hostif operational state is consistent with post damping behavior of the physical interface’s link event. If vendor SAI is running in non application mode or on a platform that doesn’t support `SAI_SWITCH_HOSTIF_OPER_STATUS_UPDATE_MODE_APPLICATION` mode, the feature will not be enabled.
+
+### SAIREDIS API
+
+Following SAI redis port attributes are proposed to configure the link event damping configurations using port SAI API:
+
+**File: sairedis.h**
+```
+/**
+ * @brief Link event damping algorithms.
+ */
+typedef enum _sai_link_event_damping_algorithm_t
+{
+    /** Link event damping algorithm disabled */
+    SAI_LINK_EVENT_DAMPING_ALGORITHM_DISABLED = 0,
+
+    /** Additive increase exponential decrease based link event damping algorithm */
+    SAI_LINK_EVENT_DAMPING_ALGORITHM_AIED = 1,
+
+} sai_link_event_damping_algorithm_t;
+
+
+typedef struct _sai_redis_link_event_damping_aied_config_t 
+{
+
+  /** Link event damping algorithm. */
+  sai_link_event_damping_algorithm_t algorithm.
+
+  /** Max link event damping suppression time. */
+  sai_uint32_t max_suppress_time;
+
+  /** Link event damping suppress threshold. */
+  sai_uint32_t suppress_threshold;
+
+  /** Link event damping reuse threshold. */
+  sai_uint32_t reuse_threshold;
+
+  /** Link event damping decay half time duration. */
+  sai_uint32_t decay_half_life;
+
+  /** Link event flap penalty. */
+  sai_uint32_t flap_penalty;
+
+} sai_redis_link_event_damping_aied_config_t;
+
+
+typedef enum _sai_redis_port_attr_t
+{
+    /**
+     * @brief Link event damping algorithm.
+     *
+     * @type sai_link_event_damping_algorithm_t
+     * @flags CREATE_AND_SET
+     * @default SAI_LINK_EVENT_DAMPING_ALGORITHM_DISABLED
+     */
+    SAI_REDIS_PORT_ATTR_LINK_EVENT_DAMPING_ALGORITHM = SAI_PORT_ATTR_CUSTOM_RANGE_START,
+
+    /**
+     * @brief Link event damping AIED configuration.
+     *
+     * @type sai_redis_link_event_damping_aied_config_t
+     * @flags CREATE_AND_SET
+     * @validonly SAI_REDIS_PORT_ATTR_LINK_EVENT_DAMPING_ALGORITHM = SAI_LINK_EVENT_DAMPING_ALGORITHM_AIED
+     * @default internal
+     */
+    SAI_REDIS_PORT_ATTR_LINK_EVENT_DAMPING_AIED_CONFIG,
```

Comment:
Is it the name of the proposed algo ? Should have "_ALGO" suffix ?

Does it make sense to define algos not in SAI_REDIS for sake of vendors who will use another implementation approach (SAI vendor layer) ?